pinc/send_mail.inc

```diff
@@ -0,0 +1,153 @@
+<?php
+include_once($relPath.'site_vars.php');
+include_once($relPath.'misc.inc'); // html_safe()
+include_once($relPath.'Project.inc');
+include_once($relPath.'User.inc');
+
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\SMTP;
+use PHPMailer\PHPMailer\Exception;
+
+function send_mail($to, $subject, $message, $from = null, $reply_to = null)
+// If this is for real, send the message.
+// If we're testing, just report what would have been sent.
+// Optional $from and $reply_to arguments may either take the form
+// "user@example.com" or "Example User <user@example.com>"
+{
+    global $testing, $phpmailer_smtp_config, $auto_email_addr, $site_name, $site_url;
+
+    $from = $from ?? $auto_email_addr;
+    $reply_to = $reply_to ?? $auto_email_addr;
+
+    // Append a standard footer to all emails sent out by the system.
+    if (!endswith($message, "\n")) {
+        $message .= "\n";
+    }
+
+    $message .= implode("\n", [
+        "",
+        mdmail_horizontal_rule(),
+        mdmail_append_linebreak($site_name),
+        $site_url,
+        "",
+        // We intentionally don't translate this message because we don't know
+        // what the recipient's language is at this point.
+        "This is an automated message.",
+    ]);
+
+    $Parsedown = new ParsedownExtra();
+    $Parsedown->setSafeMode(true);
+    $html_message = $Parsedown->text($message);
+
+    if ($testing) {
+        echo "\n<hr style='border-style:double;'>\n";
+        echo "\$testing is $testing. If it were FALSE, the following Markdown mail would have been sent:\n";
+        echo "<pre style='white-space: pre-wrap'>\n";
+        echo html_safe("To: $to\n");
+        echo html_safe("Subject: $subject\n");
+        echo html_safe("From: $from\n");
+        echo html_safe("Reply-To: $reply_to\n");
+        echo "</pre>";
+        echo "<pre style='white-space: pre-wrap'>";
+        echo html_safe("$message\n");
+        echo "</pre>\n";
+        echo "<hr style='border-style:dashed;'>\n";
+        echo "$html_message\n"   ;
+        echo "<hr style='border-style:double;'>\n";
+        // Comment out line below to allow sending of emails when testing
+        // return true;
+    }
+
+    $success = true;
+    $mail = new PHPMailer(true);
+    try {
+        // Server settings
+        $mail->isSMTP();
+        foreach ($phpmailer_smtp_config as $key => $value) {
+            $mail->$key = $value;
+        }
+
+        // Recipients
+        [$name, $address] = get_name_address($from);
+        $mail->setFrom($address, $name);
+        [$name, $address] = get_name_address($reply_to);
+        $mail->addReplyTo($address, $name);
+        $mail->addAddress($to);
+
+        // Content
+        $mail->isHTML(true);
+        $mail->Subject = $subject;
+        $mail->Body = $html_message;
+        $mail->AltBody = $message;
+
+        $mail->send();
+    } catch (Exception $e) {
+        echo "<p>Message could not be sent. Mailer Error: {$mail->ErrorInfo}</p>";
```

Comment:
I suggest we change this to a `error_log()` instead of showing an error to the user for a couple of reasons.
* The user might not even realize that what they did results in emails being sent, so this would be confusing.
* The error might contain system-level info that we don't want to show to the user -- up to and including other people's email addresses.

Something like:
```php
error_log("send_email.inc - Message could not be sent. PHPMailer Error: {$mail->ErrorInfo}");
```

Reply:
Oh yes, two good reasons!